noxfile.py

```diff
@@ -805,6 +816,40 @@ def notebook(session: nox.Session):
     for process in processes:
         process.join()
 
+    # when run via pytest, notebooks output a .bytesprocessed report
+    # collect those reports and print a summary
+    _print_bytes_processed_report()
+
+
+def _print_bytes_processed_report():
+    """Add an informational report about http queries and bytes
+    processed to the testlog output for purposes of measuring
+    bigquery-related performance changes.
+    """
+    print("---BIGQUERY USAGE REPORT---")
+    cumulative_queries = 0
+    cumulative_bytes = 0
+    for report in Path("notebooks/").glob("*/*.bytesprocessed"):
+        with open(report, "r") as f:
+            filename = os.path.basename(report)
+            filename = filename.replace(".bytesprocessed", "")
```

Comment:
nit: `filepath = os.path.dirname(filename)`

Reply:
We really do want just a filename here, but I agree basename and replace is not 100% best practice. Changed to use pathlib.Path.stem, which should be the most approved option here I think.